# The mode

## Problem 3-31
Find the mean, median, and mode for the sets (a) $3, 5, 2, 6, 5, 9, 5, 2, 8, 6$ and (b) $51.6, 48.7, 50.3, 49.5, 48.9$.

### Solution 3-31
<a id="Solution-3-31-a"></a>
(a)  Arranged in an array, the numbers are $2, 2, 3, 5, 5, 5, 6, 6, 8,$ and $9$.

\begin{align}
{\text{Mean}} &=\frac{1}{10}(2+2+3+5+5+5+6+6+8+9)=5.1 \\
{\text{Median}} &=\text{arithmetic mean of two midde numbers}=\frac{1}{2}(5+5)=5 \\
{\text{Mode}} &=\text{number occurring most frequently}=5
\end{align}

<a id="Solutio-3-31-b"></a>
(b) Arranged in an array, the numbers are $48.7, 48.9, 49.5, 50.3,$ and $51.6$
\begin{align}
{\text{Mean}} &=\frac{1}{5}(48,7+48.9+49.5+50.3+51.6)=49.8 \\
{\text{Median}} &=\text{middle number}=49.5 \\
{\text{Mode}} &=\text{number occurring most frequently (nonexistent here)}
\end{align}

In [2]:
load("../../Saves//centend.Rdata")
p3.31.a <- c(2, 2, 3, 5, 5, 5, 6, 6, 8, 9)
cat("Arranged array: ", sort(p3.31.a), "\n")
cat("Mean = ", mean(p3.31.a), "\n")
cat("Median = ", median(p3.31.a), "\n")
cat("Mode = ", mode(p3.31.a), "\n")

Arranged array:  2 2 3 5 5 5 6 6 8 9 
Mean =  5.1 
Median =  5 
Mode =  5 


In [3]:
p3.31.b <- c(51.6, 48.7, 50.3, 49.5, 48.9)
cat("Arranged array: ", sort(p3.31.b), "\n")
cat("Mean = ", mean(p3.31.b), "\n")
cat("Median = ", median(p3.31.b), "\n")
cat("Mode = ", mode(p3.31.b), "\n")

Arranged array:  48.7 48.9 49.5 50.3 51.6 
Mean =  49.8 
Median =  49.5 
Mode =  NA 


---

# Problem 3.32
Suppose you desired to find the mode for the data in Problem [3.29](05%20The%20median.ipynb#Problem-3-29). You could use frequencies procedure of SAS to obtain the following output. By considering the output from FREQ Procedure (Fig. [3-4](#Figure-3-4)), what are the modes of the alcohol-linked deaths?

In [18]:
load("../../Saves/Problem_3.29.Rdata")
p3.32.X <- unique(P3.29.data)
p3.32.f <- as.numeric(table(P3.29.data))
p3.32.percent <- round(p3.32.f/sum(p3.32.f)*100, 2)
p3.32.cumsum <- cumsum(p3.32.f)
p3.32.cumsum.percent <- cumsum(p3.32.percent)
figure.3.4 <- matrix(c(p3.32.X, p3.32.f, p3.32.percent, p3.32.cumsum, p3.32.cumsum.percent), nrow=length(p3.32.X))
colnames(figure.3.4) <- c("deaths", "Frequency", "Percent", "Cumulative Frequency", "Cumulative Percent")

#### Figure 3-4

In [19]:
figure.3.4

deaths,Frequency,Percent,Cumulative Frequency,Cumulative Percent
20,2,3.92,2,3.92
30,2,3.92,4,7.84
40,1,1.96,5,9.80
50,2,3.92,7,13.72
60,3,5.88,10,19.60
70,1,1.96,11,21.56
80,2,3.92,13,25.48
90,1,1.96,14,27.44
110,1,1.96,15,29.40
120,3,5.88,18,35.28


### Solution 3-32
The data is bimodal and the modes are $60$ and $120$. This is seen by considering the SAS output and noting that both $60$ and $120$ have a frequency of $3$ which is greater than any other of the values.

In [29]:
load("../../Saves/centend.Rdata")
p3.32.mode <- mode(P3.29.data)
cat("Mode = ", p3.32.mode, "\n")
cat("Frequency = ", figure.3.4[figure.3.4[,1]==p3.32.mode[1], 2])

Mode =  60 120 
Frequency =  3

---

# Problem 3-33
Some statistical software packages have a mode routine buit in but do not return all modes in the case that the data is multimodal. Consider the output from SPSS in Fig. [3-5](#Figure-3-5).

What does SPSS do when asked to find modes?

In [30]:
load("../../Saves/Problem_3.32.Rdata")
Figure.3.4

deaths,Frequency,Percent,Cumulative Frequency,Cumulative Percent
20,2,3.92,2,3.92
30,2,3.92,4,7.84
40,1,1.96,5,9.80
50,2,3.92,7,13.72
60,3,5.88,10,19.60
70,1,1.96,11,21.56
80,2,3.92,13,25.48
90,1,1.96,14,27.44
110,1,1.96,15,29.40
120,3,5.88,18,35.28


### Solution 3-33
SPSS prints out the smallest mode. However, it is possible to look at the frequency distribution and find all modes just as in SAS (see the output above).

---